client/lib/deterministic-stringify/index.js

```diff
@@ -1,32 +1,39 @@
-function deterministStringify( source ) {
-	var namespace = [],
-		keys;
+export function deterministicStringify( source ) {
+	if ( source === null ) {
+		return 'null';
+	}
+
+	// Handle primitive data types:
+	// boolean, number, string, undefined
 	if ( typeof source !== 'object' ) {
 		if ( typeof source === 'undefined' ) {
 			return 'undefined';
 		}
+
 		if ( typeof source === 'string' ) {
-			return "'" + source.toString() + "'";
+			return `'${ source.toString() }'`;
 		}
+
 		return source.toString();
```

Comment:
If we used `String( source )` here, the special case for `undefined` would not be necessary few lines above.